projects/v2/src/app/store/sheet.state.ts

```diff
@@ -506,6 +507,71 @@ export class SheetState {
     );
   }
 
+  /**
+   * Action to fetch the sheet data for the CSV files. Resets the Sheet State
+   * Accepts the sheet config of the particular sheet
+   */
+  @Action(FetchSheetDataFromCSV)
+  fetchSheetDataFromCSV(
```

Comment:
This is mostly copy/pasted from fetchSheetData... Find a way to have better code reuse here. Not a sustainable way to add features.